modules/sfm/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+set(the_description "SFM algorithms")
+
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Winvalid-pch)
+
+# needed since we are linking static libs
+set(CMAKE_CXX_FLAGS ${CMAKE_CSS_FLAGS} "-fPIC")
```

Comment:
doing that I get the following error:

In file included from /home/eriba/software/opencv_contrib/modules/sfm/src/libmv_light/libmv/simple_pipeline/pipeline.h:24:0,
                 from /home/eriba/software/opencv_contrib/modules/sfm/include/opencv2/sfm/simple_pipeline.hpp:47,
                 from /home/eriba/software/opencv_contrib/modules/sfm/include/opencv2/sfm.hpp:48,
                 from /home/eriba/software/opencv_contrib/modules/sfm/test/test_precomp.hpp:8,
                 from /home/eriba/software/opencv/build/modules/sfm/opencv_test_sfm_pch_dephelp.cxx:1:
/home/eriba/software/opencv_contrib/modules/sfm/src/libmv_light/libmv/simple_pipeline/callbacks.h:26:7: error: ‘class libmv::ProgressUpdateCallback’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 class ProgressUpdateCallback {
       ^
cc1plus: some warnings being treated as errors
make[2]: *** [modules/sfm/CMakeFiles/opencv_test_sfm_pch_dephelp.dir/opencv_test_sfm_pch_dephelp.cxx.o] Error 1
make[1]: *** [modules/sfm/CMakeFiles/opencv_test_sfm_pch_dephelp.dir/all] Error 2